Lib/multiprocessing/popen_spawn_posix.py

```diff
@@ -63,6 +63,6 @@ def _launch(self, process_obj):
         finally:
             if parent_r is not None:
                 self.finalizer = util.Finalize(self, os.close, (parent_r,))
-            for fd in (child_r, child_w, parent_w):
+            for fd in (child_r, child_w):
```

Comment:
I think `parent_w` needs to be registered to be close with a `Finalize` too

Reply:
As is, I am not sure. `parent_r` needs a Finalizer because it is an attribute of the `Popen` object, which may be deleted later on at some point. OTOH `parent_w` will be gc'ed at the end of this function, but we want the underlying `fd` to remain open until this process ends.

Reply:
I think @tomMoral is right. All running child processes are kept alive in the `_children` set. But when a process object is destroyed (which implies the child process has exited), we need to close the corresponding fds.

Reply:
OK. The implication process object destroyed => child process has exited was what I was missing here. Thanks for the insight.

Reply:
Not sure why the `parent_r is None` exists, but I did the same for `parent_w`.